### DISCUSSION TITLE
Is it possible to pause tile animation?

### DISCUSSION THREAD

**Discussion by ifbsj** [StarArawn/bevy_ecs_tilemap#473] [Q&A] [answered]
Hi,

I have been trying to pause the animation on a tile by using a query like this:
`mut tile_query: Query<&mut AnimatedTile>,`
And then changing the speed on some tiles to 0:
```
let entity = tile_storage.get(&TilePos { x: anchor.x, y: anchor.y });
if entity.is_some() {
     if let Ok(mut tile_anim) = tile_query.get_mut(entity.unwrap()) {
         if next_prod == 0 {
            println!("Stopping animation");
            tile_anim.speed = 0.0;
        } else {
            println!("Starting animation");
            tile_anim.speed = 1.0;
        }
    }
}
```
My printlns come out correctly, but the animation does not pause when I set speed to 0.
Is there a way to do this, other than simply despawning and respawning the tile?

**Comment by rparrett** [comment, marked as answer]
This looks like a bug. I opened a pull request to fix it here: https://github.com/StarArawn/bevy_ecs_tilemap/pull/474

You can work around it by triggering change detection on another tile property like color:

```rust
fn pause_animation(
    mut query: Query<(&mut AnimatedTile, &mut TileColor)>,
    keys: Res<Input<KeyCode>>,
) {
    if keys.just_pressed(KeyCode::P) {
        for (mut anim, mut color) in &mut query {
            anim.speed = if anim.speed == 0.0 { 1.0 } else { 0.0 };
            color.set_changed();
        }
    }
}
```

**Reply by ifbsj** [reply]
The suggested workaround works for me.
Thanks for the quick response.
